fmn/sse/sse_webserver.py

```diff
@@ -29,16 +29,16 @@ class SSEServer(resource.Resource):
     subscribers = SSESubscriber(log=logger)
 
     def render_GET(self, request):
-        request = self.add_headers(request)
-        request.write(b"")
         if self.is_valid_path(request.postpath):
+            request = self.add_headers(request)
             self.handle_request(request)
             request.notifyFinish().addBoth(self.responseFailed,
                                            request, request.postpath)
             return server.NOT_DONE_YET
         else:
+            request.setResponseCode(404)
             return self.invalid_request(request=request,
```

Comment:
Okay. The current implementation is returning a content-type of ``text/html`` rather than ``application/json``. Including more detail about what resource is missing sounds good, but right now it's just the HTTP status line repeated.

It might be good to define a subclass of ``resource.NoResource`` which includes the response template and documents the the JSON schema that is returned to the user.